go-controller/pkg/ovn/controller/services/utils.go

```diff
@@ -22,6 +22,92 @@ func deleteVIPsFromAllOVNBalancers(vips sets.String, name, namespace string) err
 	err := deleteVIPsFromNonIdlingOVNBalancers(vips, name, namespace)
 	if err != nil {
 		return errors.Wrapf(err, "Failed to delete vips from ovn balancers %s %s", name, namespace)
+		
+type lbEndpoints struct {
+	IPs  []string
+	Port int32
+}
+
+func getLocalLbEndpoints(slices []*discovery.EndpointSlice, svcPort v1.ServicePort, family v1.IPFamily, gatewayRouter string) lbEndpoints {
+	epsSet := sets.NewString()
+	lbEps := lbEndpoints{[]string{}, 0}
+	hostName := util.GetWorkerFromGatewayRouter(gatewayRouter)
+
+	// return an empty object so the caller don't have to check for nil and can use it as an iterator
+	if len(slices) == 0 {
+		return lbEps
+	}
+
+	for _, slice := range slices {
+		klog.V(4).Infof("Getting local endpoints for slice %s on node %s", slice.Name, hostName)
+		// Only return addresses that belong to the requested IP family
+		if slice.AddressType != discovery.AddressType(family) {
+			klog.V(4).Infof("Slice %s with different IP Family endpoints, requested: %s received: %s",
+				slice.Name, slice.AddressType, family)
+			continue
+		}
+	
+		// build the list of endpoints in the slice
+		for _, port := range slice.Ports {
+			// If Service port name set it must match the name field in the endpoint
+			if svcPort.Name != "" && svcPort.Name != *port.Name {
+				klog.V(5).Infof("Slice %s with different Port name, requested: %s received: %s",
+					slice.Name, svcPort.Name, *port.Name)
+				continue
+			}
+
+			// Get the targeted port
+			tgtPort := int32(svcPort.TargetPort.IntValue())
+			// If this is a string, it will return 0
+			// it has to match the port name
+			// otherwise, it has to match the port number
+			if (tgtPort == 0 && svcPort.TargetPort.String() != *port.Name) ||
+				(tgtPort > 0 && tgtPort != *port.Port) {
```

Comment:
if uint16 is used technically don't have to check if `tgtPort > 0`

Reply:
ack, this code was removed